messaginginapp/src/main/java/io/customer/messaginginapp/gist/data/listeners/Queue.kt

```diff
@@ -43,66 +43,87 @@ class Queue : GistQueue {
     private val cacheDirectory by lazy { File(application.cacheDir, "http_cache") }
     private val cache by lazy { Cache(cacheDirectory, cacheSize.toLong()) }
 
-    private fun saveToPrefs(context: Context, key: String, value: String) {
-        val prefs = context.getSharedPreferences("network_cache", Context.MODE_PRIVATE)
-        prefs.edit().putString(key, value).apply()
+    private val gistQueueService by lazy {
+        createGistQueueService()
     }
 
-    private fun getFromPrefs(context: Context, key: String): String? {
-        val prefs = context.getSharedPreferences("network_cache", Context.MODE_PRIVATE)
-        return prefs.getString(key, null)
+    private fun createGistQueueService(): GistQueueService {
+        val httpClient = OkHttpClient.Builder()
+            .cache(cache)
+            .addInterceptor { chain ->
+                val originalRequest = chain.request()
+                val networkRequest = originalRequest.newBuilder()
+                    .addHeader(NetworkUtilities.CIO_SITE_ID_HEADER, state.siteId)
+                    .addHeader(NetworkUtilities.CIO_DATACENTER_HEADER, state.dataCenter)
+                    .apply {
+                        state.userId?.let { userToken ->
+                            addHeader(
+                                NetworkUtilities.USER_TOKEN_HEADER,
+                                Base64.encodeToString(userToken.toByteArray(), Base64.NO_WRAP)
+                            )
+                        }
+                    }
+                    .header("Cache-Control", "no-cache")
+                    .build()
+
+                handleResponse(chain.proceed(networkRequest), originalRequest)
+            }
+            .build()
+
+        return Retrofit.Builder()
+            .baseUrl(GistEnvironment.PROD.getGistQueueApiUrl())
+            .addConverterFactory(GsonConverterFactory.create())
+            .client(httpClient)
+            .build()
+            .create(GistQueueService::class.java)
     }
 
-    override fun clearPrefs(context: Context) {
-        context.getSharedPreferences("network_cache", Context.MODE_PRIVATE).edit().clear().apply()
+    private fun handleResponse(response: okhttp3.Response, originalRequest: okhttp3.Request): okhttp3.Response {
```

Comment:
I think keywords like `intercept` or `transform` would be more appropriate for this method and the methods it calls. I would expect a method with `handle` prefix to be something that executes a final action. However, this method seems more focused on intercepting and transforming the response.